packages/dioxus-blitz/src/documents/dioxus_document.rs

```diff
@@ -47,26 +52,241 @@ impl Into<Document> for DioxusDocument {
     }
 }
 impl DocumentLike for DioxusDocument {
-    fn poll(&mut self, mut cx: std::task::Context) {
+    fn poll(&mut self, mut cx: std::task::Context) -> bool {
         loop {
             {
                 let fut = self.vdom.wait_for_work();
                 pin_mut!(fut);
 
                 match fut.poll_unpin(&mut cx) {
                     std::task::Poll::Ready(_) => {}
-                    std::task::Poll::Pending => break,
+                    std::task::Poll::Pending => return false,
                 }
             }
+
+            self.vdom.render_immediate(&mut MutationWriter {
+                doc: &mut self.inner,
+                state: &mut self.vdom_state,
+            });
+            println!("wrote mutatins to dom");
+            return true;
         }
     }
+
+    fn handle_event(&mut self, event: blitz_dom::events::RendererEvent) -> bool {
+        // Collect the nodes into a chain by traversing upwards
+        // This is important so the "capture" phase can be implemented
+        let mut node = event.target;
+        let mut chain = vec![node];
```

Comment:
Nit: Would be nice to preallocate a Vec of size 16 or similar.

Reply:
Nit: I've been trying to use `node_id` when it's a `NodeId` and `node` when it's a `Node`.